src/coreclr/jit/hwintrinsic.cpp

```diff
@@ -1885,6 +1885,19 @@ GenTree* Compiler::impHWIntrinsic(NamedIntrinsic        intrinsic,
                 assert(!isScalar);
                 retNode =
                     gtNewSimdHWIntrinsicNode(nodeRetType, op1, op2, op3, op4, intrinsic, simdBaseJitType, simdSize);
+
+                switch (intrinsic)
+                {
+#if defined(TARGET_ARM64)
+                    case NI_Sve_Scatter:
+                        assert(varTypeIsSIMD(op3->TypeGet()));
+                        retNode->AsHWIntrinsic()->SetAuxiliaryJitType(getBaseJitTypeOfSIMDType(sigReader.op3ClsHnd));
```

Comment:
just to confirm, we have two variants of this API. For first one, we are storing, type of `indices` as auxilary type, while for later we are storing that of `data`? 

```c#
        // <summary>
        // void svst1_scatter_[s64]offset[_f64](svbool_t pg, float64_t *base, svint64_t offsets, svfloat64_t data)
        //   ST1D Zdata.D, Pg, [Xbase, Zoffsets.D]
        // </summary>
        public static unsafe void Scatter(Vector<double> mask, double* address, Vector<long> indicies, Vector<double> data) => Scatter(mask, address, indicies, data);

        // <summary>
        // void svst1_scatter[_u64base_f64](svbool_t pg, svuint64_t bases, svfloat64_t data)
        //   ST1D Zdata.D, Pg, [Zbases.D, #0]
        // </summary>
        public static unsafe void Scatter(Vector<double> mask, Vector<ulong> addresses, Vector<double> data) 
```

looking at the codegen changes, seems we just need this for the first one? I see that we are storing auxilary type for `GatherVector` too in similar fashion, but is not needed.

Reply:
yup, we don't need the auxiliary type for the Scatter with vector of addresses (3 arg variant). Guarded it so that we only add it for the Scattter with vector of indices (4 arg variant). Did the same for gathers.